mayastor/src/subsys/opts.rs

```diff
@@ -482,15 +482,36 @@ impl GetOpts for IscsiTgtOpts {
 pub struct ErrStoreOpts {
     /// ring buffer size
     pub err_store_size: usize,
+
     /// NexusErrStore enabled
     pub enable_err_store: bool,
+
+    /// whether to fault the child due to failed re-tried IOs
+    pub fault_child_on_retry_count: bool,
+
+    /// the maximum number of errors for the same retried IO
+    pub max_retry_errors: u32,
+
+    /// whether to fault the child due to the total number of failed IOs
+    pub fault_child_on_error_count: bool,
```

Comment:
I'm not sure we want to make our lives so hard here.. Whenever an IO failed WITH retires or not, it a failed IO that has been retried. 

Put differently, an IO to LBA0 that fails, retried, fails, retried OK, is still 2 IO errors. right? The Errorstore contains more detail as to what why how *exactly* but that's more for diagnosing the core problem?

To keep it simple, I would suggest to start out simple, getting the right behavior on failed IO's to a child, would be WIN-WIN, then we can start to think about special cases

Reply:
My reasoning was that we would want to distinguish between, say, 10 different IOs which all fail on the first attempt then succeed, and one IO that fails 10 times in a row. It is easy to imagine wanting different thresholds for the two cases.
But I can take out the repeated IO error handling for now.